README.md

```diff
@@ -1 +1,253 @@
+    ```C
+    uint16_t calculate_preamble_bit_length_from_time(const uint32_t bit_rate, RAIL_RxDutyCycleConfig_t * duty_cycle_config)
+    {
+      float on_time = 0;
+      float off_time = 0;
+      float preamble_time = 0;
+      float preamle_bit_length = 0;
+
+      on_time = DUTY_CYCLE_ON_TIME;
+      off_time = duty_cycle_config->delay;
+      preamble_time = ((float)(PREAMBLE_PATTERN_LENGTH * PREAMBLE_PATTERN * PREAMBLE_OVERSAMPLING) * U_SEC) / bit_rate;
+
+      app_assert(preamble_time < on_time, "Please modify the on time according to the bitrate!\n");
+
+      while (1) {
+        preamble_time = (off_time + (2 * on_time)) / 1000000;
+        preamle_bit_length = (preamble_time * bit_rate);
+        if (preamle_bit_length <= 50000) {
+          break;
+        }
+        off_time = off_time - on_time;
+      }
+
+      if (((uint32_t)off_time) != DUTY_CYCLE_OFF_TIME) {
+        app_log_warning("Duty Cycle Off time was changed to ensure stable working\n");
+      }
+
+      return preamle_bit_length;
+    }
+    ```
+### Modify the Radio PHY configuations
+Please download the ***rail_config.c*** and ***rail_config.h*** file as the following link, and replace the original file is located in **Autogen** folder, rebuild the project and flash to the device.   
+[Optimized 9p6k radio configuation](radio_configuations/optimized_9p6k_configuation)  
+
+**Note**: Those two files will be overwritten if you change anythings in radio generator.  
+
+## Test conclusion
+### Power Consumption
+After optimzing the EM2 current and radio PHY. **Iavg = 19.036 uA**  
+<img src="images/optimized power consumption.gif">
+
+Below table is the comparison of power consumption between the sample project and the optimized project.
+|   project   |sample peoject| optimized project| 
+|:----:| :-----------:| :----------------:| 
+|**average current**| 40.674 uA| 19.036 uA |  
+
+**Conclusion:** The power consumption reduce by **53%** to **19.036 uA** after optimizing. 
+### Sensitivity
+- Rx project: Optimized long preamble duty cycle
+-  Use the [LR_Waveform_Generator](https://github.com/silabs-JimL/LR_WaveFormGenerator) to generate a waveform file and download the E4432B siganl generator. When the PER is 1%, the value at the monment represent the sensiticity.
+-  Sensitivity is **-118.4 dBm** after optimizing. Below table shows the test data.  
+  
+    |  output power    |Tx packets| Rx packets| 
+    |:----:| :-----------:| :----------------:| 
+    |-119.2 dBm| 1000 | 974 |
+    |-118.8 dBm| 1000 | 982 |
+    |-118.6 dBm| 1000 | 989 |
+    |-118.5 dBm| 1000 | 987 | 
+    |**-118.4 dBm**| **1000** | **991** |
+    |-118.3 dBm| 1000 | 994 |
+
+-  The sensitivity of original long preamble duty cycle is **-119.2 dBm**. Below table shows the test data.
```

Comment:
-  The sensitivity of the original long preamble duty cycle project is **-119.2 dBm**.